### PR TITLE
parse service account from SDS request

### PR DESCRIPTION
**Pull request by quanjielin**
https://github.com/istio/istio/issues/5903

service account passed from client will be in spiffe format, SDSS uses it to
1. construct response to client
2. construct csr template

### CODE REVIEW

File: security/pkg/nodeagent/sds/secretmanager.go [hunk 1]
```diff
@@ -18,14 +18,17 @@ import "time"
...
+	GetSecret(proxyID, serviceAccount, token string) (*SecretItem, error)
 }
 
 // SecretItem is the cached item in in-memory secret store.
 type SecretItem struct {
 	CertificateChain []byte
 	PrivateKey       []byte
 
+	// ServiceAccount passed from envoy, in spiffe format.
+	ServiceAccount string
```
Comment (incfly) [inline security/pkg/nodeagent/sds/secretmanager.go]: The service account in GCP is `foo@iam.gserviceaccount.com`, in kubernetes is like `foo`, and their corresponding spiffe id is `spiffe://accounts.google.com/foo` and `spiffe://cluster.local/ns/namespace/sa/foo`. So I think to be precise this should be `SpiffeID`?

File: security/pkg/nodeagent/sds/sdsservice.go [hunk 2]
```diff
@@ -44,6 +45,9 @@ const (
 	// credentialTokenHeaderKey is the header key in gPRC header which is used to
 	// pass credential token from envoy to SDS.
 	credentialTokenHeaderKey = "authorization"
+
+	// uriScheme is the URI scheme for Istio identities.
+	uriScheme = "spiffe"
```
Comment (incfly) [inline security/pkg/nodeagent/sds/sdsservice.go]: You can reference it here. https://github.com/istio/istio/blob/master/security/pkg/pki/util/san.go#L31

File: security/pkg/nodeagent/sds/secretmanager.go (hunk 1 above)
Comment (quanjielin) [reply security/pkg/nodeagent/sds/secretmanager.go]: renamed.

File: security/pkg/nodeagent/sds/sdsservice.go (hunk 2 above)
Comment (quanjielin) [reply security/pkg/nodeagent/sds/sdsservice.go]: updated.

**Pull request merged**